File: sdks/javascript/src/model/configured.ts
import type {ModelAdapter} from "./model-adapter.js";
import {ModelRef} from "./model-ref.js";

/** Application-facing model selection paired with its provider adapter. */
export class Model {

  /** Provider-neutral model reference consumed by the Rust runtime. */
  readonly ref: ModelRef;

  /** Adapter responsible for provider I/O. */
  readonly adapter: ModelAdapter;

  /** Creates a configured model from an internal adapter and reference. */
  constructor(ref: ModelRef, adapter: ModelAdapter) {

    this.ref = ref;
    this.adapter = adapter;

  }

}

File: sdks/python/src/orion_sdk/model/configured.py
"""Application-facing model selection paired with its provider adapter."""

from dataclasses import dataclass

from .contracts import ModelAdapter
from .types import ModelRef


@dataclass(frozen=True, slots=True)
class Model:
    """Pairs one provider model identifier with the adapter that executes it."""

    ref: ModelRef

    adapter: ModelAdapter
